### DISCUSSION TITLE
Webhook notifier issues

### DISCUSSION THREAD

**Discussion by TullingeHellMoppers** [djdd87/SynoAI#132] [General]
When running with the following Notifier, it works great!:
"Notifiers": [
    {
      "Type": "Pushbullet",
      "ApiKey": "XXXX"
    } 
  ],

But with the following Notifier the event never happens:
"Notifiers": [
    {
      "Type": "Webhook",
      "Url": "http://192.168.0.166:5000/webapi/entry.cgi?api=SYNO.SurveillanceStation.Webhook&method=Incoming&version=1&token=XXXX",
      "Method": "GET"
    } 
  ],

When trying the above Webhook, there is nothing happening and nothing at all in the log. I also tried the Webhook manually in Chrome and it works fine with GET.

Am I specifying something incorrectly?

**Comment by djdd87** [comment]
Provide your logs. Enabling debug logging may be necessary.

What is this webhook doing?

**Reply by TullingeHellMoppers** [reply]
The Webhook triggers  a recording from the Camera that had a successful object detection.

I made some  
[DebugLog_20220612.txt](https://github.com/djdd87/SynoAI/files/8886758/DebugLog_20220612.txt)
progress, enabling Debug, for whatever reason seems to have solved the crashing of the container, but there is another issue now. The SynoAI log states that the webhook executed successfully but the webhook was never executed according to the Action Event history log. 

I attached the Debug log from one of the events.

**Reply by djdd87** [reply]
If it's executing successfully, then it's executing successfully. It'll only report back success if it gets a success code from your webhook. i.e. it's outside of the control of SynoAI if your webhook isn't doing what you expect it to do.

**Comment by TullingeHellMoppers** [comment]
But the thing is that when executing the exact same webhook in a Browser it
executes fine and I also see it in the History log of the Webhook. So seems
like the Webhook trigger in SynoAI is not working, but still reporting
Success in the log.

Is it possible to see somewhere in SynoAI exactly the response coming back
from the Webhook execution? How is the Webhook call determining that it was
a success? Simply by looking for HTTP 200 or examining the response text?

**Reply by djdd87** [reply]
It really is as simple as just calling the URL and checking if the code is a success code (200-299):

![image](https://user-images.githubusercontent.com/22151993/173563814-d07671b8-b905-41ef-9a48-e3e665566e33.png)

I've just pushed a change that will include the status code for you when debug logging is enabled. See what code it's coming back with and report back.

**Reply by djdd87** [reply]
Make sure you update to the latest image to get this change first, obviously.

**Reply by djdd87** [reply]
The OP reported the solution in a different issue (#131). Here is what was posted:

> I figured out what was wrong.NOT WORKING:{      "Type": "Webhook",      "Url": "http://192.168.0.166:5000/webapi/entry.cgi?api=SYNO.SurveillanceStation.Webhook&method=Incoming&version=1",      "Method": "GET",      "Authentication": "Bearer",      "Token": "Xxxxxx",      "SendImage": "false",  "Cameras": [ "Cabin"]    }, WORKING: {      "Type": "Webhook",      "Url": "http://192.168.0.166:5000/webapi/entry.cgi?api=SYNO.SurveillanceStation.Webhook&method=Incoming&version=1&token=xxxxxxxx",      "Method": "GET",      "SendImage": "false",  "Cameras": [ "Cabin"]    },No Bearer Auth should be configured.Not sure if the ReadMe is incorrect or applies to certain conditions.But still, would be nice to see the Webhook response body in the Log, atleast when in Debug.Thanks allot for this code, now my setup detects persons and then triggers a recording. Should result in significantly less false triggers. -------- Original message --------From: djdd87 ***@***.***> Date: 6/14/22 13:24 (GMT+01:00) To: djdd87/SynoAI ***@***.***> Cc: TullingeHellMoppers ***@***.***>, Mention ***@***.***> Subject: Re: [djdd87/SynoAI] Leaving the Camera Types array blank results in a null ref exception (Issue #131) Config from discussion: "Cameras": [ { "Name": "FrontDoor", "Types": [ ], "Threshold": 30, "MinSizeX": 50, "MinSizeY": 50 } —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

There is nothing wrong with the README, it's just an example. I assume that meant the OP was using Bearer without actually requiring any authentication in their webhook and it was silently failing due to Synology's typical poor handling of standard HTTP request best practises.

It should be possible to output the webhook response and not just the status code.
